evalml/pipelines/components/transformers/encoders/target_encoder.py

```diff
@@ -84,10 +79,11 @@ def transform(self, X, y=None):
         if y is not None:
             y = infer_feature_types(y)
         X_t = self._component_obj.transform(X, y)
+
         X_t_df = pd.DataFrame(X_t, columns=X_ww.columns, index=X_ww.index)
-        return _retain_custom_types_and_initalize_woodwork(
-            X_ww.ww.logical_types, X_t_df, ltypes_to_ignore=[Categorical]
-        )
+        columns = X_ww.ww.select(exclude="categorical").columns
```

Comment:
This creates a new copy of the dataset, which seems excessive if we only care about column names.

Can you use `return_schema=True`?

Reply:
Ah, that's an interesting detail, thanks for pointing out! Will do.